components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint.util/src/main/java/org/wso2/carbon/identity/application/authentication/endpoint/util/EndpointConfigManager.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.authentication.endpoint.util;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+/**
+ * This class provides methods to handle endpoint configurations extracted
+ * from EndpointConfig.properties file
+ */
+public class EndpointConfigManager {
+
+    private static final Log log = LogFactory.getLog(EndpointConfigManager.class);
+    private static Properties prop;
+    private static String appName = null;
+    private static String appPassword = null;
+    private static String serverOrigin;
+
+    /**
+     * Initialize Tenant data manager
+     */
+    public static void init() {
+
+        prop = new Properties();
+
+        try {
+            String configFilePath = buildFilePath(Constants.TenantConstants.CONFIG_RELATIVE_PATH);
+            File configFile = new File(configFilePath);
+
+            InputStream inputStream;
+            if (configFile.exists()) {
+                log.info(Constants.TenantConstants.CONFIG_FILE_NAME + " file loaded from " + Constants
+                        .TenantConstants.CONFIG_RELATIVE_PATH);
+                inputStream = new FileInputStream(configFile);
+
+                prop.load(inputStream);
+
+            } else {
+                inputStream = EndpointConfigManager.class.getClassLoader()
+                        .getResourceAsStream(Constants.TenantConstants.CONFIG_FILE_NAME);
+                if (inputStream != null) {
+                    prop.load(inputStream);
+                    log.debug(Constants.TenantConstants.CONFIG_FILE_NAME
+                            + " file loaded from authentication endpoint webapp");
+                } else {
+                    if (log.isDebugEnabled()) {
+                        log.debug("Input stream is null while loading authentication endpoint from webapp");
```

Comment:
So this case is that the file cannot be found in either the authenticationendpoint web app class path or in the repository conf directory.
Shall we improve the debug log to convey info accordingly